plugins/woocommerce/src/Internal/DataStores/CustomMetaDataStore.php

```diff
@@ -250,9 +250,9 @@ public function get_meta_keys( $limit = 100, $order = 'ASC', $include_private =
 		$query = "SELECT DISTINCT meta_key FROM {$db_info['table']} ";
 
 		if ( ! $include_private ) {
-			$query .= $wpdb->prepare( 'WHERE meta_key != \'\' AND meta_key NOT LIKE %s ', $wpdb->esc_like( '_' ) . '%' );
+			$query .= $wpdb->prepare( "WHERE meta_key NOT BETWEEN '_' AND '_z' HAVING meta_key NOT LIKE %s ", $wpdb->esc_like( '_' ) . '%' );
 		} else {
-			$query .= "WHERE meta_key != '' ";
+			$query .= "WHERE meta_key NOT BETWEEN '_' AND '_z' ";
```

Comment:
This particular branch shouldn't be modified. When `$include_private = false` we do not want to exclude meta keys with names starting with `_` (which the `NOT BETWEEN` effectively does).

Reply:
Good catch, fixed it here: https://github.com/woocommerce/woocommerce/pull/46985/commits/b9663109c50bddead3578a01c5120e790f8feae6